### PR TITLE
Unpowered mechs can't bumpsmash

### PR DESCRIPTION
**Pull request by Super3222**
#### Changelog

:cl:
fix: Unpowered mechs can no longer bumpsmash.
/:cl:

### CODE REVIEW

File: code/game/mecha/working/ripley.dm [hunk 1]
```diff
@@ -33,6 +33,14 @@
 	cargo.Cut()
 	return ..()
 
+/obj/mecha/working/ripley/HasProximity(atom/movable/AM as mob|obj)
+	if (istype(AM, /obj/item/weapon/ore))
+		if(. && (locate(/obj/item/mecha_parts/mecha_equipment/hydraulic_clamp) in equipment))
```
Comment (Cruix) [inline code/game/mecha/working/ripley.dm]: . will always be null

**Comment by drderp3635** [comment]
but they already did

File: code/game/mecha/working/ripley.dm [hunk 2]
```diff
@@ -33,6 +33,14 @@
 	cargo.Cut()
 	return ..()
 
+/obj/mecha/working/ripley/HasProximity(atom/movable/AM as mob|obj)
```
Comment (X-TheDark) [inline code/game/mecha/working/ripley.dm]: as mob|obj is not necessary
Although I'm not sure if this is something important or not

**Comment by X-TheDark** [comment]
If this is supposed to fix **"Very Fast Gygax Running at Incredible Hihg Speed Permanently Bump-Smashing Stuff"**, then this won't fix anything, because it's the `get_charge()` proc that causes this. 

SPECIFICALLY (as in, intended behavior...blame /tg/), it will return a positive number if you have a tesla relay installed, even if you have no battery charge, which, incidentally, is also the cause of Gygax being able to Overload permanently.

**Comment by Super3222** [comment]
@X-TheDark Wow!

**Pull request closed without merging**